File: src/main/java/io/sch/historyscan/domain/contexts/analysis/clocrevisions/filesystem/ActualFileSystemTree.java
package io.sch.historyscan.domain.contexts.analysis.clocrevisions.filesystem;

import io.sch.historyscan.domain.hexagonalarchitecture.HexagonalArchitectureSPI;

@HexagonalArchitectureSPI
@FunctionalInterface
public interface ActualFileSystemTree {

    FileSystemTree from(RootFolder rootFolder);
}

File: historyscan-back/src/main/kotlin/io/sch/historyscan/domain/contexts/analysis/clocrevisions/filesystem/ActualFileSystemTree.kt
package io.sch.historyscan.domain.contexts.analysis.clocrevisions.filesystem

import io.sch.historyscan.domain.hexagonalarchitecture.HexagonalArchitectureSPI

@HexagonalArchitectureSPI
fun interface ActualFileSystemTree {
    fun from(rootFolder: RootFolder): FileSystemTree
}
